src/mono/netcore/System.Private.CoreLib/src/ILLink/ILLink.LinkAttributes.wasm.xml

```diff
@@ -116,12 +110,52 @@
       <attribute internal="RemoveAttributeInstances" />
     </type>
 
-    <!-- Microsoft.ComponentModel -->
+    <!-- System.ComponentModel -->
     <type fullname="System.ComponentModel.EditorBrowsableAttribute">
       <attribute internal="RemoveAttributeInstances" />
     </type>
   </assembly>
 
+  <!-- The following attributes are only necessary when debugging is supported -->
+  <assembly fullname="System.Private.CoreLib" feature="System.Diagnostics.Debugger.IsSupported" featurevalue="false">
+    <!-- Debugger specific attributes -->
+    <type fullname="System.Diagnostics.DebuggableAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Diagnostics.DebuggerBrowsableAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Diagnostics.DebuggerDisplayAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Diagnostics.DebuggerHiddenAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Diagnostics.DebuggerNonUserCodeAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Diagnostics.DebuggerStepperBoundaryAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Diagnostics.DebuggerStepThroughAttribute">
+      <attribute internal="RemoveAttributeInstances" />
+    </type>
+    <type fullname="System.Diagnostics.DebuggerTypeProxyAttribute">
```

Comment:
Does this unroot the proxy type as well so it can get trimmed?

Reply:
Yes